setup-test.ts

```diff
@@ -1,18 +1,31 @@
-import "@testing-library/jest-dom";
+import '@testing-library/jest-dom';
 /**
  * Handlers are automatically generated with Kubb from Open API documentation
  */
-import { handlers } from "./app/gen";
-import { setupServer } from "msw/node";
+import { handlers } from './app/gen';
+import { setupServer } from 'msw/node';
 
 window.URL.createObjectURL = vi.fn();
+Object.defineProperty(window, 'matchMedia', {
+  writable: true,
+  value: vi.fn().mockImplementation((query) => ({
```

Comment:
If I'm reading this correctly, we do this to mock browser methods and silence deck errors. I was hesitant to try this with my branch when I ran into these errors because I have concerns about "chasing errors". Basically, deck will continue to evolve and we will have to keep updating the mocks to reflect the methods invoked by deck. But, I suppose we can try this approach now and revisit it only if "chasing errors" actually becomes a real problem.

Reply:
Yeah I agree with the sentiment but this seems to be the best approach for now. Googling the error I was getting led me to [this Vitest Issue](https://github.com/vitest-dev/vitest/issues/821). It's probably fairer to say the problem is actually with `jsdom`, the DOM implementation Vitest uses under the hood. They haven't implemented every browser API, and this is the recommended way of filling missing ones as-needed.